lists-service/cicd/base/app/base.yaml

```diff
@@ -0,0 +1,85 @@
+AWSTemplateFormatVersion: '2010-09-09'
+
+Description: Base resources for the Lists service
+
+Parameters:
+  pBuild:
+    Type: String
+    Description: The build number
+  pCleanBranch:
+    Type: String
+    Description: The clean branch, can be used in resource names
+  pEnvironment:
+    Type: String
+    Description: The AWS environment this belongs to
+
+Conditions:
+
+  IsDev: !Equals
+    - !Ref pEnvironment
+    - development
+  NotProd: !Not
+    - !Equals
+      - !Ref pEnvironment
+      - production
+
+Resources:
+
+  ListsRepository:
+    Type: AWS::ECR::Repository
+    Properties:
+      EmptyOnDelete: !If [ NotProd, true, false ]
+      EncryptionConfiguration:
+        EncryptionType: AES256
+      ImageScanningConfiguration:
+        ScanOnPush: true
+      RepositoryName: !Sub
+                         - lists-${ResourceName}
+                         - ResourceName: !If [ IsDev, !Ref pCleanBranch, !Ref pEnvironment ]
+      RepositoryPolicyText:
+        Version: 2012-10-17
+        Statement:
+          - Sid: AllowPublicPull
+            Effect: Allow
+            Principal: '*'
+            Action:
+              - 'ecr:GetDownloadUrlForLayer'
+              - 'ecr:BatchGetImage'
+              - 'ecr:BatchCheckLayerAvailability'
+      Tags:
+        - Key: Environment
+          Value: !Ref pEnvironment
+        - Key: Branch
+          Value: !Ref pCleanBranch
+
+  ListsSecret:
+    Type: 'AWS::SecretsManager::Secret'
+    Properties:
+      Name: !Sub 
+              - lists-${ResourceName}
+              - ResourceName: !If [ IsDev, !Ref pCleanBranch, !Ref pEnvironment ]
+      Description: !Sub Lists app ${pEnvironment} secrets
+      GenerateSecretString: 
+        GenerateStringKey: db-password
+        PasswordLength: 12
+        ExcludeCharacters: "/@\" "
+        SecretStringTemplate: |
+          {
+            "db-password": ""
+          }
```

Comment:
An issue I've run into with this on other products is that if you update anything in the base config then it resets all the secrets which you usually dont want to do. [I just used this in userdetails](https://github.com/AtlasOfLivingAustralia/userdetails/blob/feature/containerization/cicd/base/app/base.yaml#L75-L85) to make it create the secrets on the first deploy then for subsequent deploys or updates to the base component it leaves them alone